Makefile

```diff
@@ -219,7 +219,6 @@ $(THRIFT_GEN): $(THRIFT_FILES) $(BIN)/thriftrw $(BIN)/thriftrw-plugin-yarpc
 # this needs to be both the files defining the generate command, AND the files that define the interfaces.
 $(BUILD)/generate: client/client.go encoded/encoded.go internal/internal_workflow_client.go $(BIN)/mockery
 	$Q $(BIN_PATH) go generate ./...
-	$Q touch $@
```

Comment:
if you get rid of this, it'll re-generate every single time.
just add the new-`//go:generate` file to the list of dependencies.

Reply:
It is very annoying that is materializes a file if you need to update/add a new generated file.
What is the benefit of it?

Reply:
make generate is noisy (because mockery is noisy). But it runs super fast, so I'd rather remove this make magic.

Reply:
it's normal makefile stuff, and we make changes to these mocks so _incredibly_ infrequently that it has probably already paid for itself several times over in CI and dev time.

but yea - it can be made automatic, and probably fast enough to not notice.

Reply:
On CI, it is always running from scratch, and for local, if you need to run generate, you will have to remove the file.

Reply:
because the list of dependencies hasn't been updated:
> # this needs to be both the files defining the generate command, AND the files that define the interfaces.

do that and it'll generate exactly when necessary.

I can make a PR to infer these files too.

Reply:
I mean, I don't understand `already paid for itself several times over in CI and dev time.`. 
CI runs generate every single time, and devs are not running make generate if they do not need to generate anything.

The default behavior in the mono repo or backend is to always regenerate everything, and it has many more files to generate. I don't see a big issue with regenerating here since we have a few files, and if we switch from mockery to mockgen - it will work silently.

Reply:
Anyway, I'll leave this to a separate discussion: why do we even need to optimize this?
Reverted back to touch

Reply:
CI skips this step for everything except the validation job, because it doesn't need to be done.  That job pretty much just exists to make sure stuff like this change doesn't happen.

And in dev time, correct, devs do not run `make generate`.  Because they _don't need to_ - the dependencies are wired up correctly so it generates when it might need to, and does nothing when it doesn't.

---

I think half of this is that you're looking at `make generate` in isolation instead of everything else that depends on it.  
By getting rid of the `touch $@`, this means _nothing_ that depends on code generation can be pruned by `make` - currently that includes lint/fmt/build, though not `make test` (mostly because I haven't fixed that half of the makefile yet, it's still quite janky from before and it's a lot harder because our CI pipeline depends on it in a lot of weakly-connected ways).

The server repo doesn't do this because it's so much slower and the generation dependencies are so much more complex.  There it's easier to force people to run `make go-generate` by hand as needed.  There isn't any need to do that in this repo.